pkg/services/ngalert/api/api_ruler_test.go

```diff
@@ -659,6 +659,75 @@ func TestRouteGetNamespaceRulesConfig(t *testing.T) {
 	})
 }
 
+func TestVerifyProvisionedRulesNotAffected(t *testing.T) {
+	orgID := rand.Int63()
+	group := models.GenerateGroupKey(orgID)
+	affectedGroups := make(map[models.AlertRuleGroupKey][]*models.AlertRule)
+	var allRules []*models.AlertRule
+	{
+		rules := models.GenerateAlertRules(rand.Intn(3)+1, models.AlertRuleGen(withGroupKey(group)))
+		allRules = append(allRules, rules...)
+		affectedGroups[group] = rules
+		for i := 0; i < rand.Intn(3)+1; i++ {
+			g := models.GenerateGroupKey(orgID)
+			rules := models.GenerateAlertRules(rand.Intn(3)+1, models.AlertRuleGen(withGroupKey(g)))
```

Comment:
Why randomize?

Reply:
do not like constants :) I like tests to declare how the test environment is created rather than what it looks like.